src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,244 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            Node<T> node = new Node<>(null, value, null);
+            head = node;
+            tail = node;
+            size++;
+        } else {
+            addNode(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        findIndex(index);
+        if (index == size && index != 0) {
+            addNode(value);
+        } else if (index == 0) {
+            addFirst(value);
+        } else if (size / 2 >= index) {
+            addFromHead(value, index);
+        } else {
+            addFromTail(value, index);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        findIndexGet(index);
+        return size / 2 >= index ? getFromHead(index) : getFromTail(index);
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        findIndexGet(index);
+        return size / 2 >= index ? setFromHead(value, index) : setFromTail(value, index);
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> current = findNode(index, null);
```

Comment:
maybe just findNodeByIndex and dont pass second argument to method

Reply:
not fixed